segments/models.py

```diff
@@ -83,170 +74,23 @@ def has_member(self, user):
         """
         if not user.id:
             return False
-        return self.members.filter(id=user.id).exists()
-
-    @live_sql
-    def has_member_live(self, user):
-        """
-        Live version of helper method. Return a bool indicating whether the user is a member of this segment.
-
-        Also updates the computed segment membership accordingly.
-        """
-        if not user.id:
-            return False
-        exists = False
-        if self._is_sql_based:
-            exists = bool(list(self._execute_raw_user_query(user=user)))
-        if self.static_ids:
-            exists = exists or user.id in self._parsed_static_ids
-        if self.pk:  # Verify that segment is saved
-            if exists:
-                SegmentMembership.objects.get_or_create(user=user, segment=self)
-            else:
-                SegmentMembership.objects.filter(user=user, segment=self).delete()
-        return exists
-
-    @property
-    def members(self):
-        """Return a queryset of all users (typed as settings.AUTH_USER_MODEL) that are members of this segment."""
-        # The ORM is smart enough to issue this as one query with a subquery
-        return self._users_from_ids(self.member_set.all().values_list('user_id', flat=True))
-
-    @property
-    @live_sql
-    def members_live(self):
-        """
-        Live version of .members. Issue SQL synchronously and return a raw queryset of all members.
-
-        Refreshing a segment ultimately proxies to this method.
-        """
-
-        if self._is_sql_based and not self.static_ids:
-            return self._execute_raw_user_query()
-
-        if self.static_ids and not self._is_sql_based:
-            return self._users_from_ids(self._parsed_static_ids)
-
-        if self.static_ids and self._is_sql_based:  # If there are SQL users and static users, dedupe and retrieve
-            from_sql_ids = [u.id for u in self._execute_raw_user_query()]
-            distinct_users = set(from_sql_ids + self._parsed_static_ids)
-            return self._users_from_ids(distinct_users)
+        return SegmentHelper().segment_has_member(self.id, user.id)
```

Comment:
Might we want to add the attribute to class and not create it on each function call
```
class Segment(models.Model):
    helper = SegmentHelper()
```

And then just use the object. It will be same object for all model instances
```suggestion
        return self.helper.segment_has_member(self.id, user.id)
```

Reply:
👍